sklearn/linear_model/_ridge.py

```diff
@@ -940,17 +940,24 @@ class Ridge(MultiOutputMixin, RegressorMixin, _BaseRidge):
         strength. `alpha` must be a non-negative float i.e. in `[0, inf)`.
 
         When `alpha = 0`, the objective is equivalent to ordinary least
-        squares, solved by the :class:`LinearRegression` object. For numerical
-        reasons, using `alpha = 0` with the `Ridge` object is not advised.
-        Instead, you should use the :class:`LinearRegression` object.
+        squares, solved by the :class:`LinearRegression` class. If `X.T @ X` is
+        singular (typically when `n_features > n_samples` or in the presence of
+        collinear features) then both classes should converge to the minimum
+        norm solution where the intercept does not participate in the
+        computation of the norm.
```

Comment:
```suggestion
        norm solution where the intercept does not participate in the
        computation of the norm.
```
This is my main concern/question. I would like to include it if it is possible in a consistent manner.

Reply:
If what I said in https://github.com/scikit-learn/scikit-learn/pull/25948#discussion_r1167873337 is true and confirmed by experiments, I think it would be better not to include it in the norm.

Furthermore, it might be complex to implement the "intercept-in-the-norm" formulation efficiently for all solvers (e.g. saga and co) while the data centering trick makes it trivial to get all solvers to implement the "intercept-out-of-the-norm" consistently.

Reply:
From a generalization point of view, I think that either solutions should get very similar out-of-sample performance in practice, it's more a question of maintainability and consistency with not penalizing the intercept in the ridge objective function.

Reply:
I think those concerns have been addressed in the [conclusions of the empirical study](https://github.com/scikit-learn/scikit-learn/pull/25948/#issuecomment-1528794049). Marking this thread as resolved.